src/utils/keyboard.ts

```diff
@@ -1,32 +1,93 @@
-import {Keyboard} from 'react-native';
+import {InteractionManager, Keyboard} from 'react-native';
+import getPlatform from '@libs/getPlatform';
+import CONST from '@src/CONST';
 
-let isVisible = false;
+let isNativeKeyboardVisible = false; // Native keyboard visibility
+let isWebKeyboardOpen = false; // Web keyboard visibility
+const isWeb = getPlatform() === CONST.PLATFORM.WEB;
+/**
+ * Initializes native keyboard visibility listeners
+ */
+const initializeNativeKeyboardListeners = () => {
+    Keyboard.addListener('keyboardDidHide', () => {
+        isNativeKeyboardVisible = false;
+    });
+
+    Keyboard.addListener('keyboardDidShow', () => {
+        isNativeKeyboardVisible = true;
+    });
+};
+
+/**
+ * Checks if the given HTML element is a keyboard-related input
+ */
+const isKeyboardInput = (elem: HTMLElement): boolean =>
+    (elem.tagName === 'INPUT' && !['button', 'submit', 'checkbox', 'file', 'image'].includes((elem as HTMLInputElement).type)) || elem.hasAttribute('contenteditable');
```

Comment:
We need to target specific tag elements to trigger the keyboard opening. It seems like we are missing support for the `TEXTAREA` element.

Reply:
it's done